### PR TITLE
Add missing 7.0 docs for isolated APIs in various namespaces

### PR DESCRIPTION
**Pull request by carlossanlop**
Found a few scattered APIs that were missing some documentation.

Area owners, please review:
- @dotnet/area-system-text-json 
- @dotnet/area-system-text-regularexpressions 
- @kouvel (for System.Threading)
- @dotnet/area-system-runtime (for System.Timers)
- @dotnet/area-extensions-configuration 
- @dotnet/area-system-componentmodel 
- @dotnet/area-system-io (for System.Formats.Tar)

### CODE REVIEW

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 1]
```diff
@@ -262,7 +262,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       <Docs>
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TElement">The generic definition of the element type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (gewarren) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:265]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

Comment (gewarren) [reply xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:265]: And elsewhere throughout this file...

**Comment by tarekgh** [comment]
`Microsoft.Extensions` and `System.ComponentModel` changes LGTM.

File: xml/System.Text.RegularExpressions/RegexRunner.xml [hunk 2]
```diff
@@ -1535,8 +1535,8 @@
         <Parameter Name="text" Type="System.ReadOnlySpan&lt;System.Char&gt;" Index="0" FrameworkAlternate="net-7.0" />
       </Parameters>
       <Docs>
-        <param name="text">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="text">The span of characters to scan for a pattern match.</param>
+        <summary>Used by a <see cref="T:System.Text.RegularExpressions.Regex" /> object generated by the <see cref="Overload:System.Text.RegularExpressions.Regex.CompileToAssembly" /> method.</summary>
```
Comment (stephentoub) [inline xml/System.Text.RegularExpressions/RegexRunner.xml:1539]: It's used by code generated by the source generator, not by Regex.CompileToAssembly.

File: xml/System.Threading/PeriodicTimer.xml [hunk 3]
```diff
@@ -107,7 +107,7 @@ This timer is intended to be used only by a single consumer at a time: only one
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Ensures that resources are freed and other cleanup operations are performed when the garbage collector reclaims the <see cref="T:System.Threading.PeriodicTimer" /> object.</summary>
```
Comment (stephentoub) [inline xml/System.Threading/PeriodicTimer.xml:110]: https://github.com/dotnet/runtime/pull/72454

File: xml/System.Timers/Timer.xml [hunk 4]
```diff
@@ -260,7 +260,7 @@
         <Parameter Name="interval" Type="System.TimeSpan" Index="0" FrameworkAlternate="net-7.0" />
       </Parameters>
       <Docs>
-        <param name="interval">To be added.</param>
+        <param name="interval">The period between events.</param>
```
Comment (stephentoub) [inline xml/System.Timers/Timer.xml:263]: https://github.com/dotnet/runtime/pull/72455

File: xml/System.Text.RegularExpressions/RegexRunner.xml (hunk 2 above)
Comment (carlossanlop) [reply xml/System.Text.RegularExpressions/RegexRunner.xml:1539]: Nice, thanks! I'll revert this to To be added and will let the porting tool put the correct full names.

Comment (carlossanlop) [inline xml/System.Text.RegularExpressions/RegexRunner.xml:1539]:
```suggestion
        <param name="text">To be added.</param>
        <summary>To be added.</summary>
```

**Review by eiriktsarpalis — APPROVED** [review]
System.Text.Json changes look good to me.

File: xml/System.Text.RegularExpressions/RegexRunner.xml [hunk 5]
```diff
@@ -1535,8 +1535,8 @@
...
         <remarks>To be added.</remarks>
```
Comment (carlossanlop) [inline xml/System.Text.RegularExpressions/RegexRunner.xml:1540]:
```suggestion
        <remarks>This API supports the product infrastructure and is not intended to be used directly from your code.</remarks>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 6]
```diff
@@ -310,7 +310,7 @@ This API is for use by the output of the System.Text.Json source generator and s
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TKey">The generic definition of the key type.</typeparam>
         <typeparam name="TValue">The generic definition of the value type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:313]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 7]
```diff
@@ -355,7 +355,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       <Docs>
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TElement">The generic definition of the element type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:358]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 8]
```diff
@@ -401,7 +401,7 @@ This API is for use by the output of the System.Text.Json source generator and s
...
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:404]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 9]
```diff
@@ -493,7 +493,7 @@ This API is for use by the output of the System.Text.Json source generator and s
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TKey">The generic definition of the key type.</typeparam>
         <typeparam name="TValue">The generic definition of the value type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:496]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 10]
```diff
@@ -537,7 +537,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       </Parameters>
       <Docs>
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:540]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 11]
```diff
@@ -583,7 +583,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       <Docs>
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TElement">The generic definition of the element type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:586]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 12]
```diff
@@ -627,7 +627,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       </Parameters>
       <Docs>
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:630]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 13]
```diff
@@ -673,7 +673,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       <Docs>
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TElement">The generic definition of the element type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:676]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 14]
```diff
@@ -722,7 +722,7 @@ This API is for use by the output of the System.Text.Json source generator and s
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TKey">The generic definition of the key type.</typeparam>
         <typeparam name="TValue">The generic definition of the value type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:725]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 15]
```diff
@@ -770,7 +770,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       <Docs>
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TElement">The generic definition of the element type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:773]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 16]
```diff
@@ -819,7 +819,7 @@ This API is for use by the output of the System.Text.Json source generator and s
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TKey">The generic definition of the key type.</typeparam>
         <typeparam name="TValue">The generic definition of the value type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:822]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 17]
```diff
@@ -865,7 +865,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       <Docs>
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TElement">The generic definition of the element type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:868]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 18]
```diff
@@ -911,7 +911,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       <Docs>
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TElement">The generic definition of the element type.</typeparam>
-        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:914]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 19]
```diff
@@ -951,7 +951,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       </Parameters>
       <Docs>
         <typeparam name="T">The type of the class or struct.</typeparam>
-        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to initialize the metadata with.</param>
+        <param name="options">The serialization and deserialization options to use.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:954]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 20]
```diff
@@ -1085,7 +1085,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       <Docs>
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
         <typeparam name="TElement">The generic definition of the element type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:1088]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 21]
```diff
@@ -1130,7 +1130,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       </Parameters>
       <Docs>
         <typeparam name="TCollection">The generic definition of the type.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:1133]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 22]
```diff
@@ -1217,7 +1217,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       </Parameters>
       <Docs>
         <typeparam name="T">The generic type definition.</typeparam>
-        <param name="options">To be added.</param>
+        param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:1220]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

File: xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml [hunk 23]
```diff
@@ -1411,7 +1411,7 @@ This API is for use by the output of the System.Text.Json source generator and s
       </Parameters>
       <Docs>
         <typeparam name="T">The generic definition for the enum type.</typeparam>
-        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
+        <param name="options">The serialization and deserialization options to use.</param>
```
Comment (carlossanlop) [inline xml/System.Text.Json.Serialization.Metadata/JsonMetadataServices.xml:1414]:
```suggestion
        <param name="options">The <see cref="T:System.Text.Json.JsonSerializerOptions" /> to use for serialization and deserialization.</param>
```

**Pull request merged**